cranelift/codegen/src/isa/zkasm/inst/emit.rs

```diff
@@ -589,34 +589,34 @@ impl MachInstEmit for Inst {
                 let offset = from.get_offset_with_state(state);
```

Comment:
I don’t love that `offset` is getting computed outside of match like this. I would duplicate this call in each of the uses tbh – I’d hazard a guess that if it was written that way in the first place, this wouldn’t have been a bug that required investigation :)

(Though I guess I’m also surprised that this wasn’t triggering dead code lint?)

Reply:
I think this works as intended by the `AMode` interface, as this call is already split across matches in the implementation of `AMode::get_offset_with_state`.
The same pattern is used in `riscv64` https://github.com/near/wasmtime/blob/20945d99a984c11328d8a544f2925ff0b2b79b0d/cranelift/codegen/src/isa/riscv64/inst/emit.rs#L748 although there it helps to avoid using the matches down the line altogether.
I think we can follow a similar path and have three cases for Load instruction: Register loads, Stack loads, and Global loads. In this approach, we can call `get_offset_with_state` only in the first two cases and won't need a further match statement.